libs/utils/src/hmpb/ballot_adjudication_reasons.ts

```diff
@@ -1,46 +1,50 @@
 import {
   AdjudicationReason,
   AdjudicationReasonInfo,
-  Contest,
   ContestOption,
   ContestOptionId,
   Contests,
   MarkStatus,
+  WriteInAreaStatus,
 } from '@votingworks/types';
 import { throwIllegalValue } from '@votingworks/basics';
 import { allContestOptions } from './all_contest_options';
 
 export interface Options {
-  optionMarkStatus: (option: ContestOption) => MarkStatus;
+  optionStatus: (option: ContestOption) => {
+    markStatus: MarkStatus;
+    writeInAreaStatus: WriteInAreaStatus;
+  };
 }
 
 /**
  * Enumerates all the reasons a series of contests might need adjudication.
- * Callers must provide a function that can get the mark status for any contest
- * option in the contests given.
+ * Callers must provide a function that can get the mark status and write-in
+ * area status for any contest option in the contests given.
  */
 export function* ballotAdjudicationReasons(
   contests: Contests,
-  { optionMarkStatus }: Options
+  { optionStatus }: Options
 ): Generator<AdjudicationReasonInfo> {
   if (contests.length === 0) {
     // This page is intentionally blank.
   } else {
     let isBlankBallot = true;
 
     for (const contest of contests) {
-      const selectedOptionsByContestId = new Map<
-        Contest['id'],
-        Array<{ id: ContestOption['id']; index: number }>
-      >();
+      const selectedContestOptions: Array<{
+        id: ContestOption['id'];
+        index: number;
+      }> = [];
```

Comment:
There was some useless code here, where we created a dictionary on every loop that only ever got one value in it. So I cleaned it up to be simpler.